.github/workflows/mobilecoin-workflow-dev-test.yaml

```diff
@@ -345,3 +351,139 @@ jobs:
+            --end ${{ env.END_KEYS }}
+
+    - name: Copy subset of fog keys
+      if: inputs.testing_block_v3 == 'true'
+      uses: mobilecoinofficial/gha-k8s-toolbox@v1
+      with:
+        action: toolbox-exec
+        ingest_color: ${{ inputs.ingest_color }}
+        namespace: ${{ inputs.namespace }}
+        rancher_cluster: ${{ secrets.RANCHER_CLUSTER }}
+        rancher_url: ${{ secrets.RANCHER_URL }}
+        rancher_token: ${{ secrets.RANCHER_TOKEN }}
+        command: |
+          /util/copy_account_keys.sh \
+            --src ${{ env.SRC_FOG_KEYS_DIR }} \
+            --dst ${{ env.DST_FOG_KEYS_DIR }} \
+            --start ${{ env.START_KEYS }} \
+            --end ${{ env.END_KEYS }}
+
+    - name: Test - Minting config tx
+      if: inputs.testing_block_v3 == 'true'
+      uses: mobilecoinofficial/gha-k8s-toolbox@v1
+      with:
+        action: toolbox-exec
+        ingest_color: ${{ inputs.ingest_color }}
+        namespace: ${{ inputs.namespace }}
+        rancher_cluster: ${{ secrets.RANCHER_CLUSTER }}
+        rancher_url: ${{ secrets.RANCHER_URL }}
+        rancher_token: ${{ secrets.RANCHER_TOKEN }}
+        command: |
+          /test/minting-config-tx-test.sh \
+            --token-id 1
+
+    - name: Test - Minting tx
+      if: inputs.testing_block_v3 == 'true'
+      uses: mobilecoinofficial/gha-k8s-toolbox@v1
+      with:
+        action: toolbox-exec
+        ingest_color: ${{ inputs.ingest_color }}
+        namespace: ${{ inputs.namespace }}
+        rancher_cluster: ${{ secrets.RANCHER_CLUSTER }}
+        rancher_url: ${{ secrets.RANCHER_URL }}
+        rancher_token: ${{ secrets.RANCHER_TOKEN }}
+        command: |
+          /test/minting-tx-test.sh \
+            --key-dir ${{ env.DST_KEYS_DIR }} \
+            --token-id 1
+
+    - name: Test - mobilecoind-json integration
+      if: inputs.testing_block_v3 == 'true'
+      uses: mobilecoinofficial/gha-k8s-toolbox@v1
+      with:
+        action: toolbox-exec
+        ingest_color: ${{ inputs.ingest_color }}
+        namespace: ${{ inputs.namespace }}
+        rancher_cluster: ${{ secrets.RANCHER_CLUSTER }}
+        rancher_url: ${{ secrets.RANCHER_URL }}
+        rancher_token: ${{ secrets.RANCHER_TOKEN }}
+        command: |
+          /test/mobilecoind-json-integration-test.sh \
+            --key-dir ${{ env.DST_KEYS_DIR }}
+
+    - name: Test - mint-auditor
+      if: inputs.testing_block_v3 == 'true'
+      uses: mobilecoinofficial/gha-k8s-toolbox@v1
+      with:
+        action: toolbox-exec
+        ingest_color: ${{ inputs.ingest_color }}
+        namespace: ${{ inputs.namespace }}
+        rancher_cluster: ${{ secrets.RANCHER_CLUSTER }}
+        rancher_url: ${{ secrets.RANCHER_URL }}
+        rancher_token: ${{ secrets.RANCHER_TOKEN }}
+        command: |
+          /test/mint-auditor-test.sh \
+            --token-id 1
+
+    - name: Test - use drain_accounts to transfer id 1 balances to fog keys
+      if: inputs.testing_block_v3 == 'true'
```

Comment:
Optional/Equivalent: Move this condition to the job level

Reply:
I believe we've gone over this one before. Job level conditionals to make parts of the workflow optional don't work very well in GHA.  The problem is dependency management.  For each job that depends on the job you make conditional, you have to do a nasty set of `if` statements to check if it was a failed job or skipped job.  If you have multiple dependencies, you need to do checks on all of the dependencies in the chain.  

Like just about any CI/CD system I've worked with, easy things are easy... any kind of complexity makes it 10x harder to maintain.